### ISSUE TITLE
Additional filter functionality: support NEWS.md in a subdirectory (monorepo)

### ISSUE THREAD

**Issue by dpastoor** [cynkra/fledge#9]
One case that I have is that I use a package internal to all projects/apps/etc, such that it is not the only thing being worked on within the git history tree. As such, similar to CI tags, it would be awesome if there was an option to tune what commits fledge actually scrapes for the news.

Perhaps something along the lines of `keep_if_matches` and `ignore_if_matches` as super general arguments that could be given a regular expression to test on the commit message, or even a less general [fledge-keep] or [fledge-ignore] tags

**Comment by krlmlr** [comment]
This is currently a limitation of _fledge_ that it assumes that the repo contains just one package. We could also consider a `subdir` argument, would that work for your use case? Or perhaps some logic that looks for `NEWS.md` files in the subdirectories and populates them with the corresponding log entries?

**Comment by dpastoor** [comment]
yeah package does live in its own subdir, I'd be fine with setting that and letting fledge only scrape commits where files were changed in that subdir + hunting for the news.md there.

**Comment by krlmlr** [comment]
This is a use case for a config setting where these filters can be defined. Let's revisit if/when we have config options here.

**Comment by maelle** [comment]
Related https://github.com/r-lib/gert/issues/2

**Comment by maelle** [comment]
nice defaults presented in https://github.com/cynkra/fledge/issues/625#issuecomment-1439351991 by @gadenbuie
